### PR TITLE
OpenRC service and support for more inits

### LINKED ISSUE #343: [OpenRC] Can't shutdown or reboot after login
I'm using the OpenRC script from the [ly-gentoo fork](https://github.com/pspiagicw/ly-gentoo) with LANG patch from [here](https://gist.github.com/negativeExponent/a7c3181a2c736fecf76410989f412adc).

After login it's impossible to shutdown or reboot with `loginctl poweroff` command or any other poweroff command. The only way is pushing the power button, or logging out and pressing F1 or F2.

### PR DESCRIPTION
**Pull request by Cavernosa**
I tested and it doesn't seem to break anything

Closes #343 #295

### CODE REVIEW

**Comment by wncry** [comment]
i am still trying to get this branch working on gentoo.

`start()` in `res/ly-openrc` does not execute.
i have been playing around with other commands to try to
fix this, and i will let you know when i have solved the issue.

**Comment by Cavernosa** [comment]
Weird.. if it's running openrc it should run fine since it's the exact same init and afaik it's the same structure.
You could try changing the shebang to /usr/bin/openrc-run or something, or using getty instead of agetty. (if you haven't tried these already)
That, or it's just spawning in another tty that you're not looking xd. That's all i can think of...

**Comment by gagero** [comment]
@wncry Did you manage to get it working?

**Comment by Cavernosa** [comment]
>Also I don't know if this a case specific issue (I am testing on artix linux) but I will start on tty2, no matter what the script does. So it might just be best to use tty2 (It might also be useful to note that the systemd service file uses tty2).

The script only starts ly in the provided tty, it's ly that's teleporting you to tty2, try to edit the `tty` in `config.ini` to be the same as the init. (sorry, i guess i forgot to say that)
I'll try to come with something so that the two values sync automatically.

I'm using tty7 because i thought it's easier, for the 1-6 ones you have to disable the `agetty.tty*` service so that the normal agetty doesn't override the ly agetty (assuming you have them enabled by default)

> Instead it's recommended to use [supervisor](https://wiki.gentoo.org/wiki/OpenRC/supervise-daemon#General_recipe).

Yeah, that's fine i guess.

> I made a fork with my version of the [script](https://github.com/MadcowOG/ly/blob/f05252ec77e98fef1dc7c9bcb9fd0464ab5e0570/res/ly-openrc) feel free to just steal it for this PR. I did use [this script](https://github.com/pspiagicw/ly-gentoo/blob/89a580f65fea8a35a1df64ecb92367a327c2bb79/res/ly) from [pspiagicw](https://github.com/pspiagicw) to start out because it worked somewhat.

pspiagicw's script clogs up openrc-run which introduces a lot of issues, read my issue here where i <s>tried to</s> explain: #343
Basically you shouldn't be using `exec ly` directly because it doesn't fork the process, thus clogging up openrc-run.

**Comment by MadcowOG** [comment]
I see now! I'm thinking this should be mentioned in the documentation. Although this could be intuited, some users may not intuit this.
I changed the script to fit these needs, would you say it works for the use case?

**Comment by Cavernosa** [comment]
> I see now! I'm thinking this should be mentioned in the documentation. Although this could be intuited, some users may not. 
I changed the [script](https://github.com/MadcowOG/ly/blob/openrc/res/ly-openrc) to fit these needs, would you say it works for the use case? It is essentially the same as your script, but it follows the guidelines in the [OpenRC Script Guide](https://github.com/OpenRC/openrc/blob/master/service-script-guide.md#dont-write-your-own-startstop-functions) a bit more closely. 

Yeah. Nice script, i'll try it out but i'm pretty sure it works.
One thing though, i think the TTYCONF can be boiled down to just 

`cat /etc/ly/config.ini | sed -n 's/^tty.*=[^1-9]*// p'`

> ~For automatically syncing with config.ini do you think that using sed would be appropriate?~ I got the script to pull from the config for tty. I have tested it and it appears to work.

lmao that's exactly what i was doing

**Comment by MadcowOG** [comment]
Sounds great! It might be useful to mention that the method mentioned in [#207](https://github.com/fairyglade/ly/issues/207#issuecomment-716212067) by [Figuera](https://github.com/Figuera); for changing the background and foreground color is still possible. By adding this line (Best to place it after either CONFTTY or TTY and depending on which var; the var in place of the tty will need to be changed, of course):
``` Bash
/usr/bin/printf "%b" "\e]P0{background-color}\e]P7{foreground-color}\ec" > /dev/"${TTY}"
```

**Comment by MadcowOG** [comment]
I'm recommending that along with this PR there should be a version bump so that the non-git packages also come with the fix to the issue talked about in #216 which prevents logging in, at least with systems not running systemd.

**Comment by Cavernosa** [comment]
The background color changer could be added but i don't think the init script is the best place to put that. It should be a built-in option in ly .
Also @MadcowOG i think you forgot to accept my [PR](https://github.com/MadcowOG/ly/pull/1) so i can merge the changes

**Comment by MadcowOG** [comment]
My apologies I didn't see that. I have merged it.

**Comment by MadcowOG** [comment]
> The background color changer could be added but i don't think the init script is the best place to put that. It should be a built-in option in ly .

I would like the same thing but nullgemm has said that he doesn't plan on supporting that feature, [#207](https://github.com/fairyglade/ly/issues/207#issuecomment-1110185757). So it's more of a workaround.

**Comment by MadcowOG** [comment]
One thing that is a bit annoying that could be a good change is to move the installation of each service file to their own makefile functions (like with openrc atm). Because install just always installs the systemd service file even if you are on openrc. It might require a bit of work downstream but still might be worth it for compatibility.

**Comment by MadcowOG** [comment]
I made a [new branch](https://github.com/MadcowOG/ly/tree/make-services) which contains a modified [makefile](https://github.com/MadcowOG/ly/blob/make-services/makefile) and [readme](https://github.com/MadcowOG/ly/blob/make-services/readme.md). I think this could provide better transparency over different systems.

**Comment by MadcowOG** [comment]
@wncry Would you mind testing the latest script in this PR to confirm that it works on Gentoo?

**Comment by AnErrupTion** [comment]
@MadcowOG @Cavernosa Can this be merged?

**Comment by Cavernosa** [comment]
I think so. I don't have any new ideas and the script is working fine.

**Comment by AnErrupTion** [comment]
Alright then!
